From a41e374a9c8a22c2973d403748801e1de4ba4f0e Mon Sep 17 00:00:00 2001
From: Pierre-Hugues Husson <phh@phh.me>
Date: Wed, 14 Aug 2019 23:36:45 +0200
Subject: [PATCH] Also add com.samsung.sensor.physical_proximity (if available,
 it is more a true proximity sensor than touch proximity sensor)


diff --git a/services/core/java/com/android/server/display/DisplayPowerController.java b/services/core/java/com/android/server/display/DisplayPowerController.java
index b59f6947f29..8aa68d58f41 100644
--- a/services/core/java/com/android/server/display/DisplayPowerController.java
+++ b/services/core/java/com/android/server/display/DisplayPowerController.java
@@ -553,6 +553,13 @@ final class DisplayPowerController implements AutomaticBrightnessController.Call
 
         if (!DEBUG_PRETEND_PROXIMITY_SENSOR_ABSENT) {
             mProximitySensor = mSensorManager.getDefaultSensor(Sensor.TYPE_PROXIMITY);
+            if(mProximitySensor == null) {
+                List<Sensor> sensors = mSensorManager.getSensorList(Sensor.TYPE_ALL);
+                for(Sensor sensor: sensors) {
+                    if("com.samsung.sensor.physical_proximity".equals(sensor.getStringType()))
+                        mProximitySensor = sensor;
+                }
+            }
             if(mProximitySensor == null) {
                 List<Sensor> sensors = mSensorManager.getSensorList(Sensor.TYPE_ALL);
                 for(Sensor sensor: sensors) {
-- 
2.25.1

